File: app/src/main/java/com/example/adapter/adapter/DefaultEmptyView.java
package com.example.adapter.adapter;

import android.content.Context;
import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.adapter.R;


/**
 * @author Nathaniel
 * @version V1.0.0
 * @package com.vipon.adapter
 * @datetime 2020/4/5 - 2:22
 */
public class DefaultEmptyView extends BaseEmptyView {

    private ImageView commonEmptyImage;
    private TextView commonEmptyMessage;
    private TextView commonEmptyOption;

    public DefaultEmptyView(Context context) {
        super(context);
    }

    @Override
    protected void initialize() {
        commonEmptyImage = findViewById(R.id.common_empty_image);
        commonEmptyMessage = findViewById(R.id.common_empty_message);
        commonEmptyOption = findViewById(R.id.common_empty_option);
    }

    @Override
    protected int getEmptyLayoutId() {
        return R.layout.common_empty_layout;
    }


    @Override
    protected void setEmptyOptionText(CharSequence optionText, final OnEmptyListener onEmptyListener) {
        if (TextUtils.isEmpty(optionText)) {
            return;
        }
        commonEmptyOption.setText(optionText);
        commonEmptyOption.setVisibility(VISIBLE);
        commonEmptyOption.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                if (onEmptyListener != null) {
                    onEmptyListener.onEmpty();
                }
            }
        });
    }

    @Override
    protected void setEmptyMessage(CharSequence emptyMessage) {
        if (commonEmptyMessage != null) {
            commonEmptyMessage.setText(emptyMessage);
        }
    }

    @Override
    protected void setEmptyImageResource(int drawableRes) {
        if (drawableRes != 0) {
            commonEmptyImage.setImageResource(drawableRes);
            commonEmptyImage.setVisibility(VISIBLE);
        }
    }
}

File: adapter/src/main/java/com/nathaniel/adapter/adapter/DefaultEmptyView.kt
package com.nathaniel.adapter.adapter

import android.content.Context
import android.text.TextUtils
import android.widget.ImageView
import android.widget.TextView
import com.nathaniel.adapter.R

/**
 * @author Nathaniel
 * @version V1.0.0
 * @package com.vipon.adapter
 * @datetime 2020/4/5 - 2:22
 */
open class DefaultEmptyView(context: Context?) : BaseEmptyView(context) {
    private var commonEmptyImage: ImageView? = null
    private var commonEmptyMessage: TextView? = null
    private var commonEmptyOption: TextView? = null
    override fun initialize() {
        commonEmptyImage = findViewById(R.id.common_empty_image)
        commonEmptyMessage = findViewById(R.id.common_empty_message)
        commonEmptyOption = findViewById(R.id.common_empty_option)
    }

    override val emptyLayoutId: Int
        get() = R.layout.common_empty_layout

    override fun setEmptyOptionText(optionText: CharSequence?, onEmptyListener: OnEmptyListener?) {
        if (TextUtils.isEmpty(optionText)) {
            return
        }
        commonEmptyOption!!.text = optionText
        commonEmptyOption!!.visibility = VISIBLE
        commonEmptyOption!!.setOnClickListener { onEmptyListener?.onEmpty() }
    }

    override fun setEmptyMessage(emptyMessage: CharSequence?) {
        if (commonEmptyMessage != null) {
            commonEmptyMessage!!.text = emptyMessage
        }
    }

    override fun setEmptyImageResource(drawableRes: Int) {
        if (drawableRes != 0) {
            commonEmptyImage!!.setImageResource(drawableRes)
            commonEmptyImage!!.visibility = VISIBLE
        }
    }
}